### PR TITLE
Enums and slices support

### PR DESCRIPTION
**Pull request by VisenDev**
I've added support for both slices and enums in my generic push and pop functions

### CODE REVIEW

**Comment by VisenDev** [comment]
hmm it seems not all versions of lua have the "lua_len" function, I may have to find a work around

> src/lib.zig:1170:10: error: root struct of file 'cimport' has no member named 'lua_len'

**Comment by VisenDev** [comment]
TODO: 
- finish pushSlice implementation
- audit all point processing functionality to make it less messy. Pointers have quite a bit of nuance in how they should be converted

**Comment by natecraddock** [comment]
> hmm it seems not all versions of lua have the "lua_len" function, I may have to find a work around
> 
> > src/lib.zig:1170:10: error: root struct of file 'cimport' has no member named 'lua_len'

I'll take a closer look tomorrow, but you can use `lua_rawlen` or `lua_objlen` in older versions of Lua.

I would use `lua_objlen` and `lua_rawlen` specifically in this case, because both bypass metamethods iirc

File: src/lib.zig [hunk 1]
```diff
@@ -1357,6 +1357,14 @@ pub const Lua = struct {
         return c.lua_objlen(lua.state, index);
     }
 
+    /// returns length of table in any lua version
+    pub fn versionAgnosticLen(lua: *Lua, index: i32) usize {
```
Comment (natecraddock) [inline src/lib.zig:1361]: I like the intention of this function (I meant to add one soon), but I don't love the name.

I took a look at the [LPeg](https://www.inf.puc-rio.br/~roberto/lpeg/) source code. Internally for compatibility between Lua versions Roberto (the creator of Lua) aliases rawlen to objlen. So I think we can trust that those two are compatible enough :)

So could you instead modify the `rawLen` function to call objlen for lua51 and luau? I'm also happy to do this myself after merging the PR

File: src/lib.zig [hunk 2]
```diff
@@ -3026,6 +3034,49 @@ pub const Lua = struct {
...
+            .One => {
+                const childinfo = @typeInfo(info.child).Array;
+                std.debug.assert(childinfo.child == u8);
+                std.debug.assert(childinfo.sentinel != null);
+
+                const casted: *childinfo.child = @ptrCast(@constCast(childinfo.sentinel.?));
+                if (casted.* != 0) {
+                    @compileError("Sentinel of slice must be a null terminator");
+                }
+                _ = lua.pushString(&(value.*));
```
Comment (natecraddock) [inline src/lib.zig:3061]: This `&(value.*)` is funky... is there a simpler way?

Edit: just tested, everything compiled and tested fine if I just passed `value` to pushString

**Review by natecraddock — CHANGES_REQUESTED** [review]
Just a couple things, everything else looks good!

File: src/lib.zig (hunk 2 above)
Comment (VisenDev) [reply src/lib.zig:3061]: Okay I will make this change

File: src/lib.zig (hunk 1 above)
Comment (VisenDev) [reply src/lib.zig:1361]: Sure I can do this

**Review by natecraddock — APPROVED** [review]

**Pull request merged**